toolchain/check/diagnostic_helpers.h

```diff
@@ -5,20 +5,32 @@
 #ifndef CARBON_TOOLCHAIN_CHECK_DIAGNOSTIC_HELPERS_H_
 #define CARBON_TOOLCHAIN_CHECK_DIAGNOSTIC_HELPERS_H_
 
+#include <concepts>
+#include <variant>
+
 #include "llvm/ADT/APSInt.h"
 #include "toolchain/parse/node_ids.h"
 #include "toolchain/sem_ir/ids.h"
 
 namespace Carbon::Check {
 
-// TODO: Consider instead changing calls to `SemIR::LocId::TokenOnly(...)`.
-inline auto TokenOnly(SemIR::LocId loc_id) -> SemIR::LocId {
-  return loc_id.ToTokenOnly();
-}
+class LocIdForDiagnostics {
+ public:
+  template <class T>
+    requires std::constructible_from<SemIR::LocId, T>
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  LocIdForDiagnostics(T loc_id) : loc_id_(SemIR::LocId(loc_id)) {}
+
+  // NOLINTNEXTLINE(google-explicit-constructor)
+  operator SemIR::LocId() const { return loc_id_; }
```

Comment:
Can this direction be explicit?
```suggestion
  auto loc_id() -> SemIR::LocId { return loc_id_; }
```

Reply:
Done